packages/reporting/src/patterns.js

```diff
@@ -126,6 +141,53 @@ const TRANSFORMS = new Map(
       }
     },
 
+    /**
+     * Given a URL and a list of query parameters, it returns an equivalent
+     * URL, but with those query parameters removed.
+     *
+     * Notes:
+     * - If the parameter occurs multiple times, all of them will be removed.
+     * - If the URL is invalid, null is returned.
+     *
+     * Example ["removeParams", ["foo"]]:
+     * - "https://example.test/path?foo=remove&bar=keep" -> "https://example.test/path?bar=keep"
+     * - "This is a string but not an URL" -> null
+     * - "/example.test/path" -> null (relative URLs are not supported)
+     *
+     * Example ["removeParams", ["foo", "bar"]]:
+     * - "https://example.test/path?foo=1&bar=2" -> "https://example.test/path"
+     */
+    removeParams: (x, queryParams) => {
```

Comment:
what is `x` exactly? a url string? please rename

Reply:
From the comment in the header of the primitives:
```
 * - The first argument is the current value (the accumulator),
 *   but extra parameters can be defined in the DSL; these will be
 *   passed to the function as additional arguments.
 ```
 
Not sure how to call it better. Its type is not known in advance (that is why it has to be checked at runtime). It is the result of an attribute extraction from the DOM (typically a string), but can also be the result of a transformation (which can be any type).